### PR TITLE
[ci][release] support global config for release test infra

### PR DESCRIPTION
**Pull request by can-anyscale**
Support global config for release test infra. This will allow us to control things such as BYOD ecr and test state machine data storage.

Testing:
- Unit test
- Release test

### CODE REVIEW

File: release/ray_release/configs/global_config.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
+import os
+import yaml
+
+
+class GlobalConfig:
```
Comment (aslonnie) [inline release/ray_release/configs/global_config.py:5]: maybe just use a `TypedDict`, so that you do not need to write all those getters?

Comment (can-anyscale) [reply release/ray_release/configs/global_config.py:5]: lol, I tried it but TypedDict does not allow static variable so I cannot create a singleton instance of it

**Comment by can-anyscale** [comment]
Blocked by python 3.7

**Review by aslonnie — COMMENTED** [review]
(still in draft? should I review this?)

**Review by aslonnie — APPROVED** [review]

File: release/ray_release/configs/global_config.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
+import yaml
+from typing_extensions import TypedDict
```
[the remark this answers is not in the document]
Comment (can-anyscale) [reply release/ray_release/configs/global_config.py:2]: interesting, it's already there

**Review by krfricke — APPROVED** [review]

**Pull request merged**